x/ccv/consumer/keeper/migration_test.go

```diff
@@ -0,0 +1,109 @@
+	require.NoError(t, stateStore.LoadLatestVersion())
+	registry := codectypes.NewInterfaceRegistry()
+	cdc := codec.NewProtoCodec(registry)
+	ctx := sdk.NewContext(stateStore, tmproto.Header{}, false, log.NewNopLogger())
+	require.NoError(t, stateStore.LoadLatestVersion())
+
+	// Create new empty subspace
+	subspace := paramtypes.NewSubspace(cdc,
+		codec.NewLegacyAmino(),
+		storeKey,
+		memStoreKey,
+		paramtypes.ModuleName,
+	)
+
+	// Note that new param key table is set in keeper constructor
+	subspace = subspace.WithKeyTable(v1consumertypes.ParamKeyTable())
+
+	// Set 9 params from v1.0.0
+	subspace.Set(ctx, v1consumertypes.KeyEnabled, true)
+	subspace.Set(ctx, v1consumertypes.KeyBlocksPerDistributionTransmission, int64(10))
+	subspace.Set(ctx, v1consumertypes.KeyDistributionTransmissionChannel, "channel-0")
+	subspace.Set(ctx, v1consumertypes.KeyProviderFeePoolAddrStr, "cosmos17p3erf5gv2436fd4vyjwmudakts563a497syuz")
+	subspace.Set(ctx, v1ccvtypes.KeyCCVTimeoutPeriod, time.Hour)
+	subspace.Set(ctx, v1consumertypes.KeyTransferTimeoutPeriod, time.Hour)
+	subspace.Set(ctx, v1consumertypes.KeyConsumerRedistributionFrac, "0.5")
+	subspace.Set(ctx, v1consumertypes.KeyHistoricalEntries, int64(10))
+	subspace.Set(ctx, v1consumertypes.KeyConsumerUnbondingPeriod, time.Hour)
+
+	// Confirm 3 new params cannot be set with old key table
+	require.Panics(t, func() {
+		subspace.Set(ctx, v2consumertypes.KeySoftOptOutThreshold, "0.05")
+	})
+	require.Panics(t, func() {
+		subspace.Set(ctx, v2consumertypes.KeyRewardDenoms, []string{"untrn"})
+	})
+	require.Panics(t, func() {
+		subspace.Set(ctx, v2consumertypes.KeyProviderRewardDenoms, []string{"uatom"})
+	})
+
+	// Now create new subspace, mocking an upgrade where app initialization happens again
+	subspace = paramtypes.NewSubspace(cdc,
+		codec.NewLegacyAmino(),
+		storeKey,
+		memStoreKey,
+		paramtypes.ModuleName,
+	).WithKeyTable(v2consumertypes.ParamKeyTable()) // Use v2 key table, this would be set in keeper constructor upon app init
+
+	// Run migration
+	v2consumerkeeper.MigrateParamsv1Tov2(ctx, subspace)
+
+	// Use v2 keeper to confirm params are set correctly
+	keeper := v2consumerkeeper.Keeper{}
+	keeper.SetParamSpace(ctx, subspace)
+
+	params := keeper.GetParams(ctx)
+	require.Equal(t, true, params.Enabled)
+	require.Equal(t, int64(10), params.BlocksPerDistributionTransmission)
+	require.Equal(t, "channel-0", params.DistributionTransmissionChannel)
+	require.Equal(t, "cosmos17p3erf5gv2436fd4vyjwmudakts563a497syuz", params.ProviderFeePoolAddrStr)
+	require.Equal(t, time.Hour, params.CcvTimeoutPeriod)
+	require.Equal(t, time.Hour, params.TransferTimeoutPeriod)
+	require.Equal(t, "0.5", params.ConsumerRedistributionFraction)
+	require.Equal(t, int64(10), params.HistoricalEntries)
+	require.Equal(t, time.Hour, params.UnbondingPeriod)
+	// 3 new params are set to default values
+	require.Equal(t, "0.05", params.SoftOptOutThreshold)
+	require.Equal(t, []string(nil), params.RewardDenoms)
```

Comment:
Note the default for these 2 params is a nil slice, this seems safe from a quick search. But I'm slightly suspicious of this pattern